packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -120,6 +121,49 @@ export type DirFileNameSelection = {
   outputdir: string;
 };
 
+export type TempFile = {
+  fileName: string;
+};
+
+export class CreateApexTempFile
+  implements ParametersGatherer<{ fileName: string }> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<{ fileName: string }>
+  > {
+    if (vscode.workspace.rootPath) {
+      const fileName = path.join(
+        vscode.workspace.rootPath,
+        '.sfdx',
+        'tools',
+        'tempApex.input'
+      );
+      const editor = await vscode.window.activeTextEditor;
+
+      if (!editor || editor.selection.isEmpty) {
+        return { type: 'CANCEL' };
+      }
+      const document = editor.document;
+      const writeFile = await writeFileAsync(fileName, document.getText());
```

Comment:
Maybe I'm reading https://code.visualstudio.com/docs/extensionAPI/vscode-api#TextDocument wrongly but won't document.getText() get the _whole_ text of the current active editor?

Based on what you had on the line to check for editor.selection I thought you wanted to have the selection piped to a new file?

Reply:
Good catch as I forgot to pass the selection as parameter. Done.